src/cc/arduino/plugins/wifi101/flashers/java/WINCFlasher.java

```diff
@@ -25,22 +25,92 @@
  * invalidate any other reasons why the executable file might be covered by
  * the GNU General Public License.
  */
-package cc.arduino.plugins.wifi101.flashers.java;
+ package cc.arduino.plugins.wifi101.flashers.java;
 
-import java.net.MalformedURLException;
-import java.net.URL;
-import java.security.cert.Certificate;
-import java.security.cert.X509Certificate;
-import java.util.Arrays;
-import java.util.List;
+ import java.net.MalformedURLException;
+ import java.net.URL;
+ import java.security.cert.Certificate;
+ import java.security.cert.X509Certificate;
+ import java.util.Arrays;
+ import java.util.*;
+ import java.util.List;
+ import java.io.ByteArrayOutputStream;
+ import java.io.*;
+ import java.io.File;
+ import java.io.FileInputStream;
+ import java.io.IOException;
+ import java.io.InputStream;
+ import java.net.URISyntaxException;
+ import org.apache.commons.lang3.StringUtils;
 
-import cc.arduino.plugins.wifi101.certs.WiFi101Certificate;
-import cc.arduino.plugins.wifi101.certs.WiFi101CertificateBundle;
-import cc.arduino.plugins.wifi101.firmwares.WINC1500Firmware;
-import cc.arduino.plugins.wifi101.flashers.Flasher;
+ import cc.arduino.plugins.wifi101.certs.WiFi101Certificate;
+ import cc.arduino.plugins.wifi101.certs.WiFi101CertificateBundle;
+ import cc.arduino.plugins.wifi101.flashers.Flasher;
+ import javax.swing.JProgressBar;
 
-public abstract class JavaFlasher implements Flasher {
 
+ public class WINCFlasher implements Flasher {
+
+   public String boardName;
+   public String version;
+   public File file;
+   public File Compatiblefile;
+   public JProgressBar progressBar;
+   public String name;
+   public List<String> compatibleboard;
+
+ public WINCFlasher(String _name, String _version, String _filename) {
+     boardName = _name;
+     compatibleboard =  new ArrayList<String>();
+     version = _version;
+     file = null;
+     name = "WINC1500";
+     try {
+       String jarPath = Flasher.class.getProtectionDomain().getCodeSource().getLocation().toURI().getPath();
+       File jarFolder = new File(jarPath).getParentFile();
+       file = new File(jarFolder, _filename);
+     } catch (URISyntaxException e) {
+       e.printStackTrace();
+     }
+     try {
+       String line = null;
+       String jarPath = Flasher.class.getProtectionDomain().getCodeSource().getLocation().toURI().getPath();
+       File jarFolder = new File(jarPath).getParentFile();
+       Compatiblefile = new File(jarFolder, "firmwares/Config.ini");
```

Comment:
Same applies here (a saner config file format would make this function way shorter and reliable)